hibernate-core/src/main/java/org/hibernate/procedure/internal/ParameterBindImpl.java

```diff
@@ -79,7 +79,8 @@ private void internalSetValue(T value) {
 		}
 
 		if ( procedureParameter.getParameterType() != null ) {
-			if ( !procedureParameter.getParameterType().isInstance( value ) && !procedureParameter.getHibernateType().getReturnedClass().isInstance( value ) ) {
+			if ( value != null && !procedureParameter.getParameterType().isInstance( value ) &&
```

Comment:
Currently, if the user does not do that, there will be an error like the following one:

     15:55:27,562 DEBUG SQL:94 - {call sp_is_null(?,?)}
     15:55:27,589  WARN SqlExceptionHelper:137 - SQL Error: 0, SQLState: 22023
     15:55:27,590 ERROR SqlExceptionHelper:142 - No value specified for parameter 1.

For MySQL, there will be something similar, and so on.

We could check the `isPassNullsEnabled` setting and maybe throw a `HibernateException`, maybe with a CoreMessageLogger code so that it's easier for them to track the cause.

What do you think? Should I start adding support for this too?

Reply:
Withdrawn. I had misunderstood you. Let's follow what we discussed on HipChat.